[Workflows] Core components - triggers (#4696)
elastic/docs-content: explore-analyze/workflows/triggers/alert-triggers.md

---

# Alert triggers

Alert triggers run workflows automatically when detection or alerting rules generate an alert. Use alert triggers for alert enrichment, automated incident response, case creation, or notification routing.

When an alert triggers your workflow, it provides rich context data through the `event` field.

To set up an alert trigger, follow these steps:

:::::{stepper}

::::{step} Define an alert trigger
Create a workflow with an alert trigger:

```yaml
name: Security Alert Response
description: Enriches and triages security alerts
enabled: true
triggers:
  - type: alert
steps:
  ....
```
::::

::::{step} Configure the alert rule
After creating your workflow, configure your alert rule to trigger it.

::::{tab-set}

:::{tab-item} Alerting rules

> 1. Find **{{rules-ui}}** in **{{stack-manage-app}}** or by using the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).

Reviewer: ```suggestion
1. Go to **{{rules-ui}}** in **{{stack-manage-app}}** or use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).
```

2. Find or create the alerting rule you want to trigger the workflow.

> 3. Edit the rule settings and in the **Actions** section, click **Add action**.

Reviewer: ```suggestion
3. In the rule settings, under **Actions**, select **Add action**.
```

4. Select **Workflows**.
5. Select your workflow from the dropdown or create a new one. You can only select enabled workflows.
6. Choose whether to run separate workflows for each generated alert.
7. Optionally, click **Add action** to configure multiple workflows.
8. Create or save the rule.
:::

:::{tab-item} Security detection rules

> 1. Find **Detection rules (SIEM)** in the navigation menu or by using the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).

Reviewer: ```suggestion
1. Go to **Detection rules (SIEM)** in the navigation menu or use the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md).
```

2. Find or create the detection rule you want to trigger the workflow.
3. Edit the rule settings and in the **Actions** section, select **Workflows**.

> 4. Select your workflow from the dropdown or create a new one. You can only select enabled workflows.

Reviewer: Again, it would be good to use a synonym for "enabled" here (although I'm not sure which one would be best).

Unless this is in the UI? In such a case, we should leave "enabled" as is.
